scripts/exregional_run_analysis.sh

```diff
@@ -280,6 +295,53 @@ case $MACHINE in
 
 esac
 
+else  # self cycled FV3-LAM ensemble
+    ens_nstarthr="01"  #for hourly forecast , to be adjusted
+     
+    imem=1
+    while [[ $imem -le ${NUM_ENS_MEMBERS} ]];
+      do
+      memcharv0=$( printf "%03d" $imem )
+      memchar=mem$( printf "%04d" $imem )
+
+#
+#--------------------------------------------------------------------
+#
+# Setup the INPUT directory for warm start cycles, which can be spin-up cycle or product cycle.
+#
+# First decide the source of the first guess (fg_restart_dirname) depending on cycle_type and BKTYPE:
+#  1. If cycle is spinup cycle (cycle_type == spinup) or it is the product start cycle (BKTYPE==2),
+#             looking for the first guess from spinup forecast (fcst_fv3lam_spinup)
+#  2. Others, looking for the first guess from product forecast (fcst_fv3lam)
+#
+  fg_restart_dirname=fcst_fv3lam
+#clttothink  ENKF_CYCLE_DIR should be the CYCLE_DIR in the EnKF part
+  slash_ensmem_subdir=$memchar
+  bkpathmem=${RRFSE_DIR}/${slash_ensmem_subdir}/fcst_fv3lam/INPUT
+
+#   the restart file from FV3 has a name like: ${YYYYMMDD}.${HH}0000.fv_core.res.tile1.nc
+#   But the restart files for the forecast length has a name like: fv_core.res.tile1.nc
+#   So the defination of restart_prefix needs a "." at the end.
+#
+
+   checkfile=${bkpathmem}/fv_core.res.tile1.nc
+   checkfile1=${bkpathmem}/fv_tracer.res.tile1.nc
+   if [ -r "${checkfile}" ] && [ -r "${checkfile1}" ] ; then
+     ln_vrfy ${bkpathmem}/fv_core.res.tile1.nc       fv3SAR${ens_nstarthr}_ens_mem${memcharv0}-fv3_dynvars 
+     ln_vrfy ${bkpathmem}/fv_tracer.res.tile1.nc     fv3SAR${ens_nstarthr}_ens_mem${memcharv0}-fv3_tracer 
```

Comment:
Line 327-331, how about:
```
   dynfile=${bkpathmem}/fv_core.res.tile1.nc
   tracerfile=${bkpathmem}/fv_tracer.res.tile1.nc
   if [ -r "${dynfile}" ] && [ -r "${tracerfile}" ] ; then
     ln_vrfy ${dynfile}       fv3SAR${ens_nstarthr}_ens_mem${memcharv0}-fv3_dynvars 
     ln_vrfy ${tracerfile}     fv3SAR${ens_nstarthr}_ens_mem${memcharv0}-fv3_tracer 
```

Reply:
spinup cycle is not considered in this PR, right?

Reply:
I did not see where the filelist file is generated.

Reply:
> spinup cycle is not considered in this PR, right?

Not for now.